File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dictionary = {}
        results = []

        for s in strs:
            sorted_s = sorted(s)
            sorted_s = ''.join(sorted_s)

            if sorted_s in dictionary:
                dictionary[sorted_s].append(s)
            else:
                dictionary[sorted_s] = [s]

        for k in dictionary:
            results.append(dictionary[k])

        return results

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> count = new HashMap<>();
        
        for (String s : strs) {
            // sort s
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String sorted = String.valueOf(chars);
            
            if (!count.containsKey(sorted)) {
                count.put(sorted, new ArrayList<>());
            }
            
            count.get(sorted).add(s);
        }
        
        return new ArrayList<>(count.values());
    }
}